polyfills/systemxhr/systemxhr.js

```diff
@@ -192,21 +193,61 @@
       Object.defineProperty(this, property, {
         enumerable: true,
         get: function() {
+          if (property === 'response') {
+            self._convertToResponseType();
+          }
           return this['_' + property];
         },
         set: function(value) {
           this['_' + property] = value;
-          navConnHelper.methodCall({
+          if (property !== 'responseType') {
```

Comment:
Let's assume that every request is made in plain text and then the client will convert the response to the appropriate type